src/Database/Database.php

```diff
@@ -1560,6 +2570,347 @@ public function updateDocument(string $collection, string $id, Document $documen
+     * @return void
+     * @throws AuthorizationException
+     * @throws StructureException
+     * @throws Throwable
+     */
+    private function updateDocumentRelationships(Document $collection, Document $old, Document $document): void
+    {
+        $attributes = $collection->getAttribute('attributes', []);
+
+        $relationships = \array_filter($attributes, function ($attribute) {
+            return $attribute['type'] === Database::VAR_RELATIONSHIP;
+        });
+
+        static $updateDepth = 1;
+
+        foreach ($relationships as $index => $relationship) {
+            /** @var string $key */
+            $key = $relationship['key'];
+            $value = $document->getAttribute($key);
+            $oldValue = $old->getAttribute($key);
+            $relatedCollection = $this->getCollection($relationship['options']['relatedCollection']);
+            $relationType = (string) $relationship['options']['relationType'];
+            $twoWay = (bool) $relationship['options']['twoWay'];
+            $twoWayKey = (string) $relationship['options']['twoWayKey'];
+            $side = (string) $relationship['options']['side'];
+
+            if ($oldValue == $value) {
+                $document->removeAttribute($key);
+                continue;
+            }
+
+            if ($updateDepth > Database::RELATION_MAX_DEPTH) {
+                $document->removeAttribute($key);
+
+                if ($index === \count($relationships) - 1) {
+                    return;
+                }
+
+                continue;
+            }
+
+            switch ($relationType) {
+                case Database::RELATION_ONE_TO_ONE:
+                    if (!$twoWay) {
+                        if ($value instanceof Document) {
+                            $relationId = $this->relateDocuments(
+                                $collection->getId(),
+                                $relatedCollection->getId(),
+                                $key,
+                                $document,
+                                $value,
+                                $relationType,
+                                $twoWay,
+                                $twoWayKey,
+                                $side,
+                                $updateDepth
+                            );
+                            $document->setAttribute($key, $relationId);
+                        }
+                        break;
+                    }
+
+                    switch (\gettype($value)) {
+                        case 'string':
+                            $related = $this->skipRelationships(fn () => $this->getDocument($relatedCollection->getId(), $value));
+
+                            if (
+                                $oldValue?->getId() !== $value
+                                && $this->findOne($relatedCollection->getId(), [
+                                    Query::equal($twoWayKey, [$document->getId()]),
+                                ]) !== false
```

Comment:
```suggestion
                                && $this->findOne($relatedCollection->getId(), [
                                    Query::equal($twoWayKey, [$document->getId()]),
                                ])
```

should be the same am i right?